transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/insight/SelectTransferMethodInsightTest.java

```diff
@@ -0,0 +1,166 @@
+package com.hyperwallet.android.ui.transfermethod.insight;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.matcher.ViewMatchers.assertThat;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+import android.content.Context;
+
+import androidx.test.espresso.IdlingRegistry;
+import androidx.test.rule.ActivityTestRule;
+
+import com.hyperwallet.android.ui.R;
+import com.hyperwallet.android.ui.common.insight.HyperwalletInsight;
+import com.hyperwallet.android.ui.common.repository.EspressoIdlingResource;
+import com.hyperwallet.android.ui.testutils.rule.HyperwalletExternalResourceManager;
+import com.hyperwallet.android.ui.testutils.rule.HyperwalletMockWebServer;
+import com.hyperwallet.android.ui.testutils.rule.HyperwalletTestRule;
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepositoryFactory;
+import com.hyperwallet.android.ui.transfermethod.view.SelectTransferMethodActivity;
+import com.hyperwallet.android.ui.user.repository.UserRepositoryFactory;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import java.util.Map;
+
+public class SelectTransferMethodInsightTest {
+    @ClassRule
+    public static HyperwalletExternalResourceManager sResourceManager = new HyperwalletExternalResourceManager();
+
+    @Rule
+    public HyperwalletTestRule mHyperwalletTestRule = new HyperwalletTestRule();
+    @Rule
+    public HyperwalletMockWebServer mMockWebServer = new HyperwalletMockWebServer(8080);
+    @Rule
+    public ActivityTestRule<SelectTransferMethodActivity> mActivityTestRule =
+            new ActivityTestRule<>(SelectTransferMethodActivity.class, true, false);
+    @Rule
+    public MockitoRule mMockito = MockitoJUnit.rule();
+    @Captor
+    private ArgumentCaptor<Map<String, String>> mParamsCaptor;
+
+    private HyperwalletInsight mHyperwalletInsight;
+
+    @Before
+    public void setup() {
+        mHyperwalletInsight = mock(HyperwalletInsight.class);
```

Comment:
Consider creating a rule to initiate and dispose HyperwalletInsight

Reply:
Created HyperwalletInsightMockRule